src/Coin.cpp

```diff
@@ -205,18 +210,15 @@ std::string TW::deriveAddress(TWCoinType coin, const PrivateKey& privateKey, TWD
     return TW::deriveAddress(coin, privateKey.getPublicKey(keyType), derivation);
 }
 
-std::string TW::deriveAddress(TWCoinType coin, const PublicKey& publicKey) {
-    return deriveAddress(coin, publicKey, TWDerivationDefault);
-}
-
-std::string TW::deriveAddress(TWCoinType coin, const PublicKey& publicKey, TWDerivation derivation) {
+std::string TW::deriveAddress(TWCoinType coin, const PublicKey& publicKey, TWDerivation derivation, const std::string& hrp) {
     auto p2pkh = TW::p2pkhPrefix(coin);
-    const auto* hrp = stringForHRP(TW::hrp(coin));
-
+    const char* hrpRaw = [&hrp, coin]() {
+        return hrp.empty() ? stringForHRP(TW::hrp(coin)) : hrp.c_str();
+    }();
     // dispatch
     auto* dispatcher = coinDispatcher(coin);
     assert(dispatcher != nullptr);
-    return dispatcher->deriveAddress(coin, derivation, publicKey, p2pkh, hrp);
+    return dispatcher->deriveAddress(coin, derivation, publicKey, p2pkh, hrpRaw);
```

Comment:
Possible memory leak here, e.g. look at [Nervos address construction](https://github.com/trustwallet/wallet-core/blob/4993b9af0d76140ffe6fa52a6a624c87afe04257/src/Nervos/Address.cpp): the hrp raw string is not copied. So when exit the scope of this function, `hrp` will be de-allocated and nervos address will potentially have some garbage data.

Reply:
We are using memory leak at each run and there is no leak according to the sanitizer, also deriveAddress is a view only function, the input parameters are only use to produce the output, here an Address.

Reply:
Valid risk, but not memory leak, but worse, use-after-free.
However, any function function taking a `char*` parameter can assume that it is valid while the function is executing, but if it needs to store it, it should copy it.
So I think passing here a `const char*` which is coming either from input parameter or from `stringForHrp` (which was the case so far), is OK.